### PR TITLE
Resolve #1136 -- LevelUp now uses StatsModifier

### LINKED ISSUE #1136: Mana stat modifier is applied again on each level up.
Whenever a champion levels up any mana gained from items is applied again on top of already existing mana modifiers.

### PR DESCRIPTION
**Pull request by cabeca1143**
-Changed direct stat changes to StatsModifiers.
-Fixed Total Mana being added to BaseValue Every level up.

Resolve #1136

### CODE REVIEW

File: GameServerLib/GameObjects/Stats/Stats.cs [hunk 1]
```diff
@@ -214,23 +214,22 @@ public void LevelUp()
...
-            ManaPoints.BaseValue = ManaPoints.Total + ManaPerLevel;
-            CurrentMana = ManaPoints.Total / (ManaPoints.Total - ManaPerLevel) * CurrentMana;
-            AttackDamage.BaseValue += AdPerLevel;
-            Armor.BaseValue += ArmorPerLevel;
-            MagicResist.BaseValue += MagicResistPerLevel;
-            HealthRegeneration.BaseValue += HealthRegenerationPerLevel;
-            ManaRegeneration.BaseValue += ManaRegenerationPerLevel;
-
+            StatsModifier statsLevelUp = new StatsModifier();
+            statsLevelUp.HealthPoints.BaseBonus = HealthPerLevel;
```
Comment (danil179) [inline GameServerLib/GameObjects/Stats/Stats.cs:218]: Should it be `=` or did you mean `+=`? (as all the others are `+=`)
Alternatively, shouldn't all the others be `=` as you only declare the variable? (I assume the default value is 0?)

**Review by danil179 — COMMENTED** [review]
Some comments.

File: GameServerLib/GameObjects/Stats/Stats.cs (hunk 1 above)
Comment (cabeca1143) [reply GameServerLib/GameObjects/Stats/Stats.cs:218]: Oh, it was supposed to be just =
Functionally = or += doesn't change anything in this case, but I'll be uploading the changes anyway.

**Review by danil179 — APPROVED** [review]
Approved. I'm not sure if that resolves the issue as I didn't check. In case it doesn't a new issue will be submitted.

**Review by danil179 — CHANGES_REQUESTED** [review]
Cannot merge. Tests fail:
```
LeagueSandbox.GameServerTests.Tests.TestStats.TestLevelUp() in C:\projects\gameserver\GameServerLibTests\Tests\TestStats.cs:line 62
```

**Review by danil179 — APPROVED** [review]
Approved. Didn't test. Should solve most of the problems. I suggest to check where in the code `BaseValue` is used and see if it should be affected by leveling up.

**Comment by danil179** [comment]
Refs #1144.

**Pull request merged**